src/lib/hasher/hasher.ts

```diff
@@ -168,14 +174,82 @@ export function getLeafHasher<V extends CCIPVersion = CCIPVersion>({
   destChainSelector,
   onRamp,
   version,
-}: Lane<V>): LeafHasher {
+}: Lane<V>): LeafHasher<V> {
   switch (version) {
     case CCIPVersion.V1_2:
     case CCIPVersion.V1_5:
-      return getV12LeafHasher(sourceChainSelector, destChainSelector, onRamp)
+      return getV12LeafHasher(sourceChainSelector, destChainSelector, onRamp) as LeafHasher<V>
```

Comment:
No need to cast here, your functions should be able to return the versioned LeafHashers on their own

Reply:
They do, but `getLeafHasher` does not know if the function is actually returning the same version `V`